dali/kernels/test/static_switch_test.cc

```diff
@@ -69,3 +70,40 @@ TEST(StaticSwitch, Nested) {
   // check that the test functor was actually called
   EXPECT_EQ(calls, 4) << "Test functor was expected to be called 4 times";
 }
+
+#include "dali/pipeline/data/types.h"  // NOLINT
+
+TEST(StaticSwitch, DALIDataType) {
+  using namespace dali;  // NOLINT
+
+  std::array<DALIDataType, 7> type_ids = {
+    DALI_UINT8,
+    DALI_INT16,
+    DALI_INT32,
+    DALI_INT64,
+    DALI_FLOAT,
+    DALI_FLOAT64,
+    DALI_BOOL
+  };
+
+  std::array<const std::type_info *, 7> type_infos  = {
+    &typeid(uint8_t),
+    &typeid(int16_t),
+    &typeid(int32_t),
+    &typeid(int64_t),
+    &typeid(float),
+    &typeid(double),
+    &typeid(bool)     // NOLINT
+  };
+
+  for (size_t i = 0; i < type_ids.size(); i++) {
```

Comment:
I guess I'm missing that, but where `id2type` is tested?

Reply:
No, you're not missing that.
It's not that critical, either, because it's not used in TYPE_SWITCH and doesn't need to satisfy any API or sth.

Reply:
Adding just:
```
EXPECT_EQ(typeid(id2type<type_ids[i]>::value), typeid(type));
```
won't do?

Reply:
No, it won't - it needs a compile-time value. We'd need a VALUE_SWITCH to do that.